geps/gep-1619/index.md

```diff
@@ -317,71 +346,55 @@ In this section, we will explore the questions and design elements associated wi
+// Support: Extended
+type LoadBalancerPolicy struct {
     metav1.TypeMeta   `json:",inline"`
     metav1.ObjectMeta `json:"metadata,omitempty"`
 
-    // Spec defines the desired state of SessionPersistencePolicy.
-    Spec SessionPersistencePolicySpec `json:"spec"`
+    // Spec defines the desired state of LoadBalancerPolicy.
+    Spec LoadBalancerPolicySpec `json:"spec"`
 
-    // Status defines the current state of SessionPersistencePolicy.
-    Status SessionPersistencePolicyStatus `json:"status,omitempty"`
+    // Status defines the current state of LoadBalancerPolicy.
+    Status LoadBalancerPolicyStatus `json:"status,omitempty"`
 }
 
-// SessionPersistencePolicySpec defines the desired state of
-// SessionPersistencePolicy.
+// LoadBalancerPolicySpec defines the desired state of
+// LoadBalancerPolicy.
 // Note: there is no Override or Default policy configuration.
-type SessionPersistencePolicySpec struct {
+type LoadBalancerPolicySpec struct {
     // TargetRef identifies an API object to apply policy to.
-    // The TargetRef may be a Service, HTTPRoute, GRPCRoute,
-    // or a HTTPRouteRule or GRPCRouteRule section.
-    // At least one of these targets must be supported for
-    // core-level compliance.
-    //
+    // Currently, Services are the only valid API target references.
     TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
 
-    // AbsoluteTimeoutSeconds defines the absolute timeout of the
-    // persistent session measured in seconds. Once
-    // AbsoluteTimeoutSeconds has elapsed, the session becomes invalid.
-    //
-    // Support: Core
-    //
-    // +optional
-    AbsoluteTimeoutSeconds int64 `json:"absoluteTimeoutSeconds,omitempty"`
-
-    // IdleTimeoutSeconds defines the idle timeout of the
-    // persistent session measured in seconds. Once the session
-    // has been idle for more than specified IdleTimeoutSeconds
-    // duration, the session becomes invalid.
-    //
-    // Support: Core
-    //
-    // +optional
-    IdleTimeoutSeconds int64 `json:"idleTimeoutSeconds,omitempty"`
+    // SessionPersistence defines and configures session persistence.
+    SessionPersistence *SessionPersistence `json:"sessionPersistence"`
+}
 
+// SessionPersistence defines the desired state of
+// SessionPersistence.
+type SessionPersistence struct {
     // SessionName defines the name of the persistent session token
     // (e.g. a cookie name).
     //
-    // Support: Extended
+    // Support: Implementation-specific
     //
     // +optional
     // +kubebuilder:validation:MaxLength=4096
     SessionName String `json:"sessionName,omitempty"`
 }
 
-// SessionPersistencePolicyStatus defines the observed state of SessionPersistencePolicy.
-type SessionPersistencePolicyStatus struct {
-    // Conditions describe the current conditions of the SessionPersistencePolicy.
+// LoadBalancerPolicyStatus defines the observed state of LoadBalancerPolicy.
+type LoadBalancerPolicyStatus struct {
```

Comment:
I think this needs to be expanded to allow for multiple controllers to write to this. I'd recommend reusing what @candita and @youngnick did for BackendTLSPolicy: https://github.com/kubernetes-sigs/gateway-api/blob/1f7b47f3acb137def7d2ebafc1e3a5b6c710252c/apis/v1alpha2/policy_types.go#L187